src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -46,6 +46,11 @@ internal abstract class AnalyzerDriver : IDisposable
         private Task _primaryTask;
 
         /// <summary>
+        /// Number of worker tasks processing compilation events and executing analyzer actions.
+        /// </summary>
+        private const int _workerCount = 5;
```

Comment:
Not sure whats a good number here. I randomly attempted with 5 and 10 and found it gave better perf with 5 in my local runs. It probably needs more perf measurements to fine tune this, but I thought this is good enough to unblock the throughput issues for now.